src/lib.rs

```diff
@@ -44,93 +66,118 @@ fn is_promise_success() -> bool {
     }
 }
 
+impl Default for LinkDrop {
+    fn default() -> Self {
+        let empty_set: LookupSet<PublicKey> = LookupSet::new(0);
+        Self {
+            accounts: empty_set,
+            nft_contract_id: String::from(""),
+        }
+    }
+}
+
 #[near_bindgen]
 impl LinkDrop {
+    #[init]
+    pub fn init(nft_contract_id: AccountId) -> Self {
+        Self {
+            accounts: LookupSet::new(0),
+            nft_contract_id: nft_contract_id,
+        }
+    }
+
     /// Allows given public key to claim sent balance.
     /// Takes ACCESS_KEY_ALLOWANCE as fee from deposit to cover account creation via an access key.
     #[payable]
     pub fn send(&mut self, public_key: Base58PublicKey) -> Promise {
-        assert!(
-            env::attached_deposit() > ACCESS_KEY_ALLOWANCE,
-            "Attached deposit must be greater than ACCESS_KEY_ALLOWANCE"
-        );
         let pk = public_key.into();
-        let value = self.accounts.get(&pk).unwrap_or(0);
-        self.accounts.insert(
-            &pk,
-            &(value + env::attached_deposit() - ACCESS_KEY_ALLOWANCE),
-        );
-        Promise::new(env::current_account_id()).add_access_key(
-            pk,
-            ACCESS_KEY_ALLOWANCE,
-            env::current_account_id(),
-            b"claim,create_account_and_claim".to_vec(),
-        )
+        let new_account = self.accounts.insert(&pk);
+        // If the set did not have this value present, true is returned
+        if new_account {
+            Promise::new(env::current_account_id()).add_access_key(
+                pk,
+                ACCESS_KEY_ALLOWANCE,
+                env::current_account_id(),
+                // add_access_key allows given pk to call functions claim or create_account_and_claim
+                b"claim,create_account_and_claim".to_vec(),
+            )
+        } else {
+            panic!("Account already registered");
+        }
     }
```

Comment:
What do you think if we use an assert?
```suggestion
        assert!(!new_account, "Account already registered");
        Promise::new(env::current_account_id()).add_access_key(
              pk,
              ACCESS_KEY_ALLOWANCE,
              env::current_account_id(),
              // add_access_key allows given pk to call functions claim or create_account_and_claim
              b"claim,create_account_and_claim".to_vec(),
          )
```

Reply:
Fixed: [Commit](https://github.com/Shard-Labs/near-linkdrop/pull/1/commits/ba20ac597220379d4cba8c9ee1401a20e1ab6fda)